baybe/surrogates/base.py

```diff
@@ -56,13 +59,39 @@ class Surrogate(ABC, SerialMixin):
     """Class variable encoding whether or not the surrogate supports transfer
     learning."""
 
+    _input_transform = field(init=False, default=None, eq=False)
+    """Callable preparing surrogate inputs for training/prediction.
+
+    Transforms a dataframe containing parameter configurations in experimental
+    representation to a corresponding dataframe containing their computational
+    representation. Only available after the surrogate has been fitted."""
+
+    _target_transform = field(init=False, default=None, eq=False)
+    """Callable preparing surrogate targets for training.
+
+    Transforms a dataframe containing target measurements in experimental
+    representation to a corresponding dataframe containing their computational
+    representation. Only available after the surrogate has been fitted."""
+
     def to_botorch(self) -> Model:
         """Create the botorch-ready representation of the model."""
         from baybe.surrogates._adapter import AdapterModel
 
         return AdapterModel(self)
 
-    def posterior(self, candidates: Tensor) -> tuple[Tensor, Tensor]:
+    def transform_inputs(self, data: pd.DataFrame) -> pd.DataFrame:
+        """Transform an experimental parameter dataframe."""
+        if self._input_transform is None:
+            raise ModelNotTrainedError("The model must be trained first.")
+        return self._input_transform(data)
+
+    def transform_targets(self, data: pd.DataFrame) -> pd.DataFrame:
+        """Transform an experimental measurement dataframe."""
+        if self._target_transform is None:
+            raise ModelNotTrainedError("The model must be trained first.")
+        return self._target_transform(data)
+
+    def posterior(self, candidates: pd.DataFrame) -> tuple[Tensor, Tensor]:
```

Comment:
Docstring needs to be adjusted (DataFrame vs. Tensor)

Reply:
Yes, forgot, because that is exactly the part that is not yet finished (here, the output type will be changed completely). So I have now simply removed the docstring details and we'll have to add them back once these details are clear 🙃